compose/material/material/src/desktopMain/kotlin/androidx/compose/material/DesktopMenu.desktop.kt

```diff
@@ -228,29 +296,78 @@ fun CursorDropdownMenu(
     expandedStates.targetState = expanded
 
     if (expandedStates.currentState || expandedStates.targetState) {
-        val transformOriginState = remember { mutableStateOf(TransformOrigin.Center) }
+        OpenDropdownMenu(
+            expandedStates = expandedStates,
+            popupPositionProvider = rememberCursorPositionProvider(),
+            onDismissRequest = onDismissRequest,
+            focusable = focusable,
+            modifier = modifier,
+            content = content
+        )
+    }
+}
 
-        var focusManager: FocusManager? by mutableStateOf(null)
-        var inputModeManager: InputModeManager? by mutableStateOf(null)
+/**
+ * Represents the open/closed state of a dropdown menu.
+ */
+@Stable
+class DropdownMenuState(initialStatus: Status = Status.Closed) {
 
-        Popup(
-            focusable = focusable,
-            onDismissRequest = onDismissRequest,
-            popupPositionProvider = rememberCursorPositionProvider(),
-            onKeyEvent = {
-                handlePopupOnKeyEvent(it, onDismissRequest, focusManager!!, inputModeManager!!)
-            },
-        ) {
-            focusManager = LocalFocusManager.current
-            inputModeManager = LocalInputModeManager.current
-
-            DropdownMenuContent(
-                expandedStates = expandedStates,
-                transformOriginState = transformOriginState,
-                modifier = modifier,
-                content = content
-            )
+    /**
+     * The current status of the menu.
+     */
+    var status: Status by mutableStateOf(initialStatus)
+
+    @Immutable
+    sealed class Status {
+
+        class Open(val position: IntOffset) : Status() {
+
+            override fun equals(other: Any?): Boolean {
+                if (this === other)
+                    return true
+
+                if (other !is Open)
+                    return false
+
+                if (position != other.position)
+                    return false
+
+                return true
```

Comment:
`return position == other.position` can be used without previous `if` statement

Reply:
Usually yes, but in `equals`, the template is to check all the fields for non-equality, one by one, and at the end return `true`. Handling the last field in a different way means more lines changed when you add a new field.

On the other hand, this class is unlikely to ever get new fields, so I think both options are OK.